velox/functions/lib/Re2Functions.h

```diff
@@ -50,6 +49,8 @@ enum class PatternKind {
   kRelaxedSuffix,
   /// Patterns matching '%{c0}%', such as '%foo%%', '%%%hello%'.
   kSubstring,
+  /// Patterns matching '%{c0}%{c1}%', such as '%%foo%%bar%%', '%foo%bar%'.
+  kSubstrings,
```

Comment:
Why do we need both kSubstring and kSubstrings ? Can we combine these two?

Reply:
just for respect old code.

Reply:
Let's combine these options.

Reply:
Update : kSubstring is for non-constant case now.

I will add benchmark to measure determinePatternKind(), and disscuss it again,  I simple assume that parseSubstrings will add perf regression and only add it for constant-case.